src/main/kotlin/uk/gov/gdx/datashare/queue/HmppsQueueFactory.kt

```diff
@@ -0,0 +1,140 @@
+package uk.gov.gdx.datashare.queue
+
+import com.amazon.sqs.javamessaging.ProviderConfiguration
+import com.amazon.sqs.javamessaging.SQSConnectionFactory
+import com.amazonaws.services.sns.model.SubscribeRequest
+import com.amazonaws.services.sqs.AmazonSQS
+import com.amazonaws.services.sqs.model.CreateQueueRequest
+import com.amazonaws.services.sqs.model.QueueAttributeName
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.context.ConfigurableApplicationContext
+import org.springframework.jms.config.DefaultJmsListenerContainerFactory
+import javax.jms.Session
+
+class HmppsQueueFactory(
+  private val context: ConfigurableApplicationContext,
+  private val amazonSqsFactory: AmazonSqsFactory,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun createHmppsQueues(hmppsSqsProperties: HmppsSqsProperties, hmppsTopics: List<HmppsTopic> = listOf()) =
+    hmppsSqsProperties.queues
+      .map { (queueId, queueConfig) ->
+        val sqsDlqClient = getOrDefaultSqsDlqClient(queueId, queueConfig, hmppsSqsProperties)
+        val sqsClient = getOrDefaultSqsClient(queueId, queueConfig, hmppsSqsProperties, sqsDlqClient)
+          .also { subscribeToLocalStackTopic(hmppsSqsProperties, queueConfig, hmppsTopics) }
```

Comment:
Perhaps just a convention I've not seen before, but I'm a little surprised by all these uses of `.also` without reference to `it`